tests/gui_tests/test_download_data.py

```diff
@@ -0,0 +1,1260 @@
+    assert widget.is_downloading is False
+
+
+def test_compose_method():
+    """Test the compose method creates correct widgets."""
+    # Skip this test as it requires app context for Textual widgets
+    # The compose method is tested indirectly through the app context tests
+    pass
+
+
+def test_watch_files_downloaded_without_errors():
+    """Test watching files_downloaded changes without errors."""
+    widget = DownloadData()
+    widget.error_files = 0
+    widget.total_files = 10
+
+    with patch.object(widget, "query_one") as mock_query:
+        mock_label = MagicMock()
+        mock_query.return_value = mock_label
+
+        widget.watch_files_downloaded(5)
+
+        mock_label.update.assert_called_once_with("Files: 5/10")
+
+
+def test_watch_files_downloaded_with_errors():
+    """Test watching files_downloaded changes with errors."""
+    widget = DownloadData()
+    widget.error_files = 2
+    widget.total_files = 10
+
+    with patch.object(widget, "query_one") as mock_query:
+        mock_label = MagicMock()
+        mock_query.return_value = mock_label
+
+        widget.watch_files_downloaded(5)
+
+        mock_label.update.assert_called_once_with("Files: 5/10 (❌ Errors: 2)")
+
+
+def test_watch_error_files_first_error():
+    """Test watching error_files when first error occurs."""
+    widget = DownloadData()
+    widget.files_downloaded = 3
+    widget.total_files = 10
+
+    with patch.object(widget, "query_one") as mock_query:
+        mock_label = MagicMock()
+        mock_query.return_value = mock_label
+
+        widget.watch_error_files(1)
+
+        mock_label.update.assert_called_once_with("Files: 3/10 (❌ Errors: 1)")
+
+
+def test_watch_error_files_subsequent_errors():
+    """Test watching error_files for subsequent errors."""
+    widget = DownloadData()
+    widget.files_downloaded = 3
+    widget.total_files = 10
+
+    with patch.object(widget, "query_one") as mock_query:
+        mock_label = MagicMock()
+        mock_query.return_value = mock_label
+
+        widget.watch_error_files(2)
+
+        mock_label.update.assert_called_once_with("Files: 3/10 (❌ Errors: 2)")
+
+
+def test_watch_is_downloading_enables_button():
+    """Test watching is_downloading changes enables button."""
+    widget = DownloadData()
+    widget.selected_project_id = "test-project"
+
+    with patch.object(widget, "query_one") as mock_query:
+        mock_button = MagicMock()
+        mock_query.return_value = mock_button
+
+        widget.watch_is_downloading(False)
+
+        mock_button.disabled = False
+
+
+def test_watch_is_downloading_disables_button():
+    """Test watching is_downloading changes disables button."""
+    widget = DownloadData()
+    widget.selected_project_id = "test-project"
+
+    with patch.object(widget, "query_one") as mock_query:
+        mock_button = MagicMock()
+        mock_query.return_value = mock_button
+
+        widget.watch_is_downloading(True)
+
+        mock_button.disabled = True
```

Comment:
shouldn't these have some validation, like `assert mock_button.disabled is False/True`